services/contract_call_local.proto

```diff
@@ -58,6 +58,23 @@ message ContractLoginfo {
     bytes data = 4;
 }
 
+/**
+ * Info about a contract account's nonce value.
+ * A nonce of a contract is only incremented when that contract creates another contract.
+ */
+message ContractNonceInfo {
```

Comment:
I'm a little confused about this type appearing in the contract_call_local.proto. Does it also apply to normal contract calls?

Reply:
It is in this file because of `ContractFunctionResult` (it is used only there) and the idea was to keep related types close together.
The comment on `ContractFunctionResult` states that:
```
This is part of the response to a ContractCallLocal query, and is in the record for a ContractCall or ContractCreateInstance transaction.
```
so yes, as `ContractNonceInfo` is part of `ContractFunctionResult` it will apply to normal contract calls as well.

Reply:
I think the issue is that `ContractFunctionResult` should likely not have been in `contract_call_local.proto`.
Placing `ContractNonceInfo` here seems to be a colocation attempt for readability since `ContractFunctionResult` is the intended consumer of it.

Another option would be `basic_types.proto` or a new custom file for Contract related messages.

@rbair23 does this current location work or would you prefer it in another location as suggested?

Reply:
Btw now I noticed [this comment](https://github.com/hashgraph/hedera-protobufs/pull/257#discussion_r1117668864) and it seems that initially we had it in `basic_types.proto` but we moved it here.

Reply:
@MiroslavGatsanoga can you move this to a new `contract_types.proto` file.
I think that should address concerns and we can get this PR in to unblock HIP 729 development.
In a future protobuf version we can explore moving more contract types into this file